### PR TITLE
Add TfLite micro ExpandDims reference kernel

### PR DESCRIPTION
**Pull request by sicong-li-arm**
Add TfLite micro ExpandDims reference kernel and tests (int8, uint8, float32)

Signed-off-by: SiCongLi <sicong.li@arm.com>

### CODE REVIEW

**Comment by gmiodice** [comment]
Hi, would it be possible to have the review for this PR?

**Review by petewarden — APPROVED** [review]

File: tensorflow/lite/micro/kernels/all_ops_resolver.cc [hunk 1]
```diff
@@ -70,6 +70,7 @@ AllOpsResolver::AllOpsResolver() {
   AddBuiltin(BuiltinOperator_RELU, Register_RELU());
   AddBuiltin(BuiltinOperator_RELU6, Register_RELU6());
   AddBuiltin(BuiltinOperator_MEAN, Register_MEAN());
+  AddBuiltin(BuiltinOperator_EXPAND_DIMS, Register_EXPAND_DIMS());
```
Comment (rthadur) [inline tensorflow/lite/micro/kernels/all_ops_resolver.cc]: `ld.lld: error: undefined symbol: tflite::ops::micro::Register_EXPAND_DIMS()
>>> referenced by all_ops_resolver.cc:73 (third_party/tensorflow/lite/micro/kernels/all_ops_resolver.cc:73)
>>>               blaze-out/k8-asan-fastbuild/bin/third_party/tensorflow/lite/micro/kernels/_objs/all_ops_resolver/all_ops_resolver.pic.o:(tflite::ops::micro::AllOpsResolver::AllOpsResolver())
clang: error: linker command failed with exit code 1 (use -v to see invocation)`

**Review by rthadur — CHANGES_REQUESTED** [review]
pasted internal error , can you please check ?

**Comment by gbaned** [comment]
@sicong-li-arm Can you please check rthadur's comments and keep us posted? Thanks!

File: tensorflow/lite/micro/kernels/all_ops_resolver.cc (hunk 1 above)
Comment (sicong-li-arm) [reply tensorflow/lite/micro/kernels/all_ops_resolver.cc]: Hi @rthadur . Sorry for the late reply. Could you please share with me how you produced this error, like the build command you used, the toolchain (which version of clang), OS etc, as I wasn't able to re-produce this linker error? Thanks!

Comment (rthadur) [reply tensorflow/lite/micro/kernels/all_ops_resolver.cc]: @petewarden can you please assist with this ?

**Comment by sicong-li-arm** [comment]
Had a discussion with @njeffrie. Closing this as it's not required by most tflite models.

**Pull request closed without merging**